pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/values.star

```diff
@@ -8,3 +10,14 @@ if hasattr(values.kappController, 'namespace') and values.kappController.namespa
 else:
     kappNamespace = values.namespace
 end
+
+def generateBashCmdForDNS(serviceCIDR):
+    regexp.match('^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)([/][0-3][0-2]?|[/][1-2][0-9]|[/][0-9])?$', serviceCIDR) or assert.fail("Please provider a valid service CIDR")
```

Comment:
I understand you are trying to get this done creatively in the template itself but don't think there is an option if you want it to be bug free and maintainable.  Additionally this logic would not work when serviceCIDR is ipv6. Lets bite the bullet and move the retrieval of coreDNSIP to CLI layer.